ResearchKit/Common/ORKHeightPicker.m

```diff
@@ -0,0 +1,272 @@
+/*
+ Copyright (c) 2016, Ricardo Sánchez-Sáez.
+ 
+ Redistribution and use in source and binary forms, with or without modification,
+ are permitted provided that the following conditions are met:
+ 
+ 1.  Redistributions of source code must retain the above copyright notice, this
+ list of conditions and the following disclaimer.
+ 
+ 2.  Redistributions in binary form must reproduce the above copyright notice,
+ this list of conditions and the following disclaimer in the documentation and/or
+ other materials provided with the distribution.
+ 
+ 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+ may be used to endorse or promote products derived from this software without
+ specific prior written permission. No license is granted to the trademarks of
+ the copyright holders even if such marks are included in this software.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#import "ORKHeightPicker.h"
+#import "ORKResult_Private.h"
+#import "ORKAnswerFormat_Internal.h"
+#import "ORKAccessibilityFunctions.h"
+
+
+@interface ORKHeightPicker () <UIPickerViewDataSource, UIPickerViewDelegate>
+
+@end
+
+
+@implementation ORKHeightPicker {
+    UIPickerView *_pickerView;
+    ORKHeightAnswerFormat *_answerFormat;
+    id _answer;
+    __weak id<ORKPickerDelegate> _pickerDelegate;
+}
+
+@synthesize pickerDelegate = _pickerDelegate;
+
+- (instancetype)initWithAnswerFormat:(ORKHeightAnswerFormat *)answerFormat answer:(id)answer pickerDelegate:(id<ORKPickerDelegate>)delegate {
+    self = [super init];
+    if (self) {
+        NSAssert([answerFormat isKindOfClass:[ORKHeightAnswerFormat class]], @"answerFormat should be ORKHeightAnswerFormat");
+        
+        // Take into account locale changes so unit string can be updated
+        [[NSNotificationCenter defaultCenter] addObserver:self
+                                                 selector:@selector(currentLocaleDidChange:)
+                                                     name:NSCurrentLocaleDidChangeNotification object:nil];
+
+        self.answer = answer;
+        _answerFormat = answerFormat;
```

Comment:
I'm passing the `answerFormat` because this implementation takes into account the case in which the user changes the locale while she is looking at the question midway through running the app ([this line](https://github.com/rsanchezsaez/ResearchKit/blob/rsanchezsaez-ORKHeightAnswerFormat/ResearchKit/Common/ORKAnswerFormat.m#L2394)), so it didn't want to implement that logic twice.